launcher/src/main/java/de/itemis/mps/gradle/launcher/MpsBackendBuilder.java

```diff
@@ -89,7 +90,7 @@ public MpsBackendBuilder withJvmVendorSpec(JvmVendorSpec jvmVendorSpec) {
 
     public MpsBackendBuilder withJavaExecutable(Provider<String> javaExecutable) {
         this.javaLauncher.set(javaExecutable
-                .map(path -> SpecificInstallationToolchainSpec.fromJavaExecutable(objects, path))
+                .map(path -> SpecificInstallationToolchainSpec.fromJavaExecutable(propertyFactory, path))
```

Comment:
This change will make this method incompatible with 8.12 and below, but apparently no-one cares because nobody complained about it not working with 8.13 either.

For the record, it could be made compatible with a reflection-based approach like this: https://github.com/specificlanguages/mps-gradle-plugin/blob/5dfceb38ae7cfc1e921147f21668ff72fbcfc214/subprojects/jbr-toolchain/src/main/kotlin/com/specificlanguages/jbrtoolchain/ToolchainSpecFactory.kt

but maybe all those `withJvm` and `withJava` methods should be deprecated and removed instead because they don't seem to bring enough value.